### DISCUSSION TITLE
What is the difference between `torch.Tensor([1,2,3])` and `torch.tensor([1,2,3])`

### DISCUSSION THREAD

**Discussion by shivammavihs** [mrdbourke/pytorch-deep-learning#277] [Q&A]
I want to understand what is the significance of each function `torch.Tensor([1,2,3])` and `torch.tensor([1,2,3])`.

The one difference I found is torch.Tensor() creates tensors with int64 dtype and torch.tensor() creates float32 dtype by default. Is there any other significant difference between both.

And, when and where to use which one?

**Comment by mrdbourke** [comment]
Hi @shivammavihs,

I didn't even know they created tensors in different data types!

I'd say I use `torch.tensor()` far more often.

`float32` is a far more often used datatype throughout machine learning/deep learning too.

**Comment by kartik-powar-3103** [comment]
Hi @shivammavihs 

The torch.tensor infers data type automatically whereas torch.Tensor assigns the default data type

I think better explanation can be found in this discussion forum:
https://discuss.pytorch.org/t/difference-between-torch-tensor-and-torch-tensor/30786/2
